### PR TITLE
Update buckets token overlay to use generic SelectorList component

### LINKED ISSUE #13912: Create new copy of BuilderCard for Token ReadWrite list
Renaming buildercard to selectorlist for better use in the token ui to select and deselect buckets.

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #13912 

Describe your proposed changes here.
Created a copy of buildercard component in shared folder to make it more generic for selecting from a list in ui

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: ui/src/shared/components/selectorList/SelectorListBody.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// Components
+import {DapperScrollbars} from '@influxdata/clockface'
+
+interface Props {
+  scrollable: boolean
+  addPadding: boolean
+  testID: string
+}
+
+export default class BuilderCardBody extends PureComponent<Props> {
```
Comment (jademcgough) [inline ui/src/shared/components/selectorList/SelectorListBody.tsx]: Should `BuilderCardBody` be `SelectorListBody`?

File: ui/src/shared/components/selectorList/SelectorListEmpty.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Libraries
+import React, {PureComponent} from 'react'
+
+interface Props {
+  testID: string
+}
+
+export default class BuilderCardEmpty extends PureComponent<Props> {
```
Comment (jademcgough) [inline ui/src/shared/components/selectorList/SelectorListEmpty.tsx]: Should `BuilderCardEmpty` be `SelectorListEmpty`?

**Comment by jademcgough** [comment]
There's an existing `SelectorList` component in `timeMachine/components/SelectorList.tsx`. Is it weird to have two components with the same name?

Now that `BuilderCard` has been made into a shared component (`SelectorList`), can `BuilderCard` in TimeMachine be replaced with the shared component?

File: ui/src/shared/components/selectorList/SelectorListBody.tsx (hunk 1 above)
Comment (Palakp41) [reply ui/src/shared/components/selectorList/SelectorListBody.tsx]: I updated it

File: ui/src/shared/components/selectorList/SelectorListEmpty.tsx (hunk 2 above)
Comment (Palakp41) [reply ui/src/shared/components/selectorList/SelectorListEmpty.tsx]: I updated it

**Comment by Palakp41** [comment]
The `SelectorList` in `timeMachine` folder is still using `BuilderCard` component. I want to keep SelectorList to be an independent and generic component. That's a good catch, thanks :)

**Pull request merged**